tests/adapters/test_epics_adapter.py

```diff
@@ -0,0 +1,130 @@
+import sys
+from dataclasses import is_dataclass
+
+import pytest
+from mock import MagicMock, Mock, create_autospec
+from mock.mock import call, mock_open, patch
+
+from tickit.core.adapter import ConfigurableAdapter, Interpreter
+from tickit.core.device import Device
+
+# % % % % % % % % % % WARNING % % % % % % % % % %
+# sys.modules["softioc"] = Mock() must be located
+# above the import of tickit.adapters.epicsadapter
+sys.modules["softioc"] = Mock()
+from tickit.adapters.epicsadapter import (  # noqa; isort:skip
+    EpicsAdapter,
+    InputRecord,
+    softioc,
+)
```

Comment:
Would this work as a `patch` context?

Reply:
You probably could do it that way, if you nested them about 8 deep. I was having difficulties getting it to work so I switched strategies to this instead.

Reply:
If you meant could you do:
`with patch("softioc") as etc:`
I don't think so, it was the first thing I tried but it didn't work for me.

Reply:
What if we were to do something along the lines of? Not even sure that'd work but it seems like it should
```python
@pytest.fixture(scope='session', autouse=True)
def epics_imports():
    import softioc
    with mock.patch.object(softioc) as softioc:
        from tickit.adapters.epicsadapter import EpicsAdapter, InputRecord
```

Reply:
I'll test it locally, that would be very useful if it does work.

Reply:
`tests/adapters/test_epics_adapter.py EEEEEEEEE                                             [100%]`
`E           AttributeError: module 'softioc' has no attribute 'rsplit'`

Yeah I guess it was too much to hope for that it would just work

Reply:
It's an error with the `patch` method itself, not with the code we're testing.

Reply:
@callumforrester any suggestions?

Reply:
This method of mocking the module has been removed.